wordpress-mobile/WordPress-Android
# [Bloganuary] Implement Dashboard Card and Modal for Bloganuary Nudge

Fixes #19714

This branch is a feature branch used for development of the full Bloganuary Nudge feature, which was divided into several other PRs that were merged into this one:
- https://github.com/wordpress-mobile/WordPress-Android/pull/19664
- https://github.com/wordpress-mobile/WordPress-Android/pull/19673
- https://github.com/wordpress-mobile/WordPress-Android/pull/19677
- https://github.com/wordpress-mobile/WordPress-Android/pull/19684
- https://github.com/wordpress-mobile/WordPress-Android/pull/19704
- https://github.com/wordpress-mobile/WordPress-Android/pull/19705
- https://github.com/wordpress-mobile/WordPress-Android/pull/19706
- https://github.com/wordpress-mobile/WordPress-Android/pull/19707

Just a reminder that we are targeting the `release/23.8` branch because this needs to get into December's release as aligned here: pcdRpT-4FE-p2#comment-7532

-----

## To Test:
All interactions with the card and modal send analytics tracking events, for testing them properly check the appropriate Test instructions in the Analytics PRs above.

- Install and log into the Jetpack app with an account/site that has Prompts available
- Go to `My Site` dashboard
- **Verify** the Bloganuary Returns card is shown
- If prompts is currently turned on
- Tap "Learn More"
- **Verify** the modal opens with the appropriate content (different when Prompts is ON and OFF)
- For Prompts ON
   - **Verify** all modal actions simply dismiss the modal
 - For Prompts OFF
   - Tap the "Turn on blogging prompts" button
   - **Verify** the modal is dismissed and the dashboard is refreshed, showing the Prompts card
 - Finally, tap the "Hide this" context menu item
 - **Verify** the card is hidden for this site (check other sites to see the card is still there)

There's no way for the user to bring the card back, only by clearing the app settings or logging out.

-----

## Regression Notes

1. Potential unintended areas of impact

    - Affecting other dashboard cards.

2. What I did to test those areas of impact (or what existing automated tests I relied on)

    - Manual Testing

3. What automated tests I added (or what prevented me from doing so)

    - Several unit tests were updated and new unit tests were created for new code.

-----

## PR Submission Checklist:


-----

## UI Changes Testing Checklist:
